packages/lib-classifier/src/hooks/useSessionStorage.js

```diff
@@ -0,0 +1,20 @@
+import { addDisposer, onSnapshot } from 'mobx-state-tree'
+import { useEffect, useState } from 'react'
+
+import { asyncSessionStorage } from '@helpers'
+
+function persist(storageKey, store, storage) {
+  function _saveSnapshot(snapshot) {
+    const data = JSON.stringify(snapshot)
+    storage.setItem(storageKey, data)
+  }
+  const snapshotDisposer = onSnapshot(store, _saveSnapshot)
+  return addDisposer(store, snapshotDisposer)
+}
+
+export default function useSessionStorage(enableStorage, store, storageKey) {
+  if (!!store && enableStorage) {
+    persist(storageKey, store, asyncSessionStorage)
+  }
```

Comment:
Something to check: Does this need an additional check to make sure that the `onSnapshot` handler isn't added if it's already been added to the store? I'm not sure if this function can be run more than once.

Reply:
Turns out that `ClassifierContainer` renders more than once, running the hook multiple times.